L1Trigger/L1TTrackMatch/plugins/L1TkMuonProducer.cc

```diff
@@ -670,6 +718,61 @@ void L1TkMuonProducer::build_tkMuons_from_idxs(TkMuonCollection& tkMuons,
     l1tkmu.setTrackCurvature(matchTk.rInv());
     l1tkmu.setTrkzVtx((float)tkv3.z());
     l1tkmu.setMuonDetector(detector);
+    l1tkmu.setQuality(l1muRef->hwQual());
+
+    // apply region cleaning (probably this is not the best way, but since this is going to
+    // be a patch and temporary, it is OK)
+    if (useRegionEtaMatching_) {
+      if (detector == barrel_MTF_region) {
+        if (std::abs(l1tkmu.eta()) > etaBO_)
+          continue;
+      } else if (detector == overlap_MTF_region) {
+        if (std::abs(l1tkmu.eta()) < etaBO_)
+          continue;
+        if (std::abs(l1tkmu.eta()) > etaOE_)
+          continue;
+      } else if (detector == endcap_MTF_region) {
+        if (std::abs(l1tkmu.eta()) < etaOE_)
+          continue;
+      }
+    }
+    tkMuons.push_back(l1tkmu);
+  }
+  return;
+}
+
+void L1TkMuonProducer::build_tkMuons_from_idxs(TkMuonCollection& tkMuons,
+                                               const std::vector<int>& matches,
+                                               const edm::Handle<L1TTTrackCollectionType>& l1tksH,
+                                               const edm::Handle<EMTFTrackCollection>& emtfTksH,
+                                               int detector) const {
+  for (uint imatch = 0; imatch < matches.size(); ++imatch) {
+    int match_trk_idx = matches[imatch];
+    if (match_trk_idx < 0)
+      continue;  // this muon was not matched to any candidate
+
+    // take properties of the track
+    const L1TTTrackType& matchTk = (*l1tksH.product())[match_trk_idx];
+    const auto& p3 = matchTk.momentum();
+    const auto& tkv3 = matchTk.POCA();
+    float p4e = sqrt(mu_mass * mu_mass + p3.mag2());
+    math::XYZTLorentzVector l1tkp4(p3.x(), p3.y(), p3.z(), p4e);
+
+    edm::Ptr<L1TTTrackType> l1tkPtr(l1tksH, match_trk_idx);
+
+    auto l1emtfTrk =
+        emtfTksH.isValid() ? edm::Ref<EMTFTrackCollection>(emtfTksH, imatch) : edm::Ref<EMTFTrackCollection>();
+
+    float trkisol = -999;
+    TkMuon l1tkmu(l1tkp4, l1emtfTrk, l1tkPtr, trkisol);
+    l1tkmu.setTrackCurvature(matchTk.rInv());
+    l1tkmu.setTrkzVtx((float)tkv3.z());
```

Comment:
is the extra float cast necessary? (c-style casts should be avoided in general, code rule 4.15 http://cms-sw.github.io/cms_coding_rules.html)